core/storage/changes_trie/impl/storage_changes_tracker_impl.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef KAGOME_STORAGE_CHANGES_TRIE_STORAGE_CHANGES_TRACKER_IMPL
+#define KAGOME_STORAGE_CHANGES_TRIE_STORAGE_CHANGES_TRACKER_IMPL
+
+#include "storage/changes_trie/changes_tracker.hpp"
+
+namespace kagome::storage::changes_trie {
+
+  class StorageChangesTrackerImpl : public ChangesTracker {
+   public:
+    enum class Error { EXTRINSIC_IDX_GETTER_UNINITIALIZED = 1 };
+
+    /**
+     * Functor that returns the current extrinsic index, which is supposed to
+     * be stored in the state trie
+     */
+    StorageChangesTrackerImpl();
+    ~StorageChangesTrackerImpl() override = default;
+
+    void setConfig(const ChangesTrieConfig &conf) override;
+
+    void setExtrinsicIdxGetter(GetExtrinsicIndexDelegate f) override;
```

Comment:
I mean, can we avoid it without making the Tracker depend on Storage? And more than that, on the current persistent batch.

Reply:
This was the simplest solution I'd came up with. I can make a method that would update the extrinsic index and call it before calling onChange everywhere, but I don't think it would be better.

Reply:
Got it. Yeah, I also don't know how to do otherwise